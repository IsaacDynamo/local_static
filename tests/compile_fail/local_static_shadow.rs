// Static mut variables transformed by `local_static` should have behavior similar to regular static mut variables.
#![deny(warnings)]

use local_static::local_static;

#[local_static]
fn main() {
    static mut X: u32 = 0;
    *X += 1;
    static mut X: u32 = 0; // ERROR
    *X += 1;
}
