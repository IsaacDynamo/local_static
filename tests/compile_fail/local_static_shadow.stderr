error: the name `X` is defined multiple times
  --> tests/compile_fail/local_static_shadow.rs:10:16
   |
10 |     static mut X: u32 = 0; // ERROR
   |                ^
